src/Controls/src/Core/Handlers/Items/Android/RecyclerViewScrollListener.cs

```diff
@@ -60,17 +60,30 @@ public override void OnScrolled(RecyclerView recyclerView, int dx, int dy)
 			if (Last == -1)
 				return;
 
-			switch (_itemsView.RemainingItemsThreshold)
+			var remainingItemsThreshold = _itemsView.RemainingItemsThreshold;
+			var itemsSourceCount = ItemsViewAdapter.ItemsSource.Count;
+			var hasHeader = ItemsViewAdapter.ItemsSource.HasHeader;
+			var hasFooter = ItemsViewAdapter.ItemsSource.HasFooter;
+
+			// Adjust the logical item count (data items only) to align with the adjusted indices returned by GetVisibleItemsIndex
+			var logicalItemCount = itemsSourceCount - (hasHeader ? 1 : 0) - (hasFooter ? 1 : 0);
+
+			switch (remainingItemsThreshold)
 			{
 				case -1:
 					return;
 				case 0:
-					if (Last == ItemsViewAdapter.ItemsSource.Count - 1)
+					if (Last == logicalItemCount - 1)
+					{
 						_itemsView.SendRemainingItemsThresholdReached();
+					}
 					break;
 				default:
-					if (ItemsViewAdapter.ItemsSource.Count - 1 - Last <= _itemsView.RemainingItemsThreshold)
+					var remainingItems = logicalItemCount - 1 - Last;
```

Comment:
Here, needs to ensure Last is within valid range. If `Last` is invalid or greater than `logicalItemCount`, remainingItems could be negative, causing incorrect threshold triggers. Thinking in cases with a race condition, empty collection with header/footer etc.